### PR TITLE
kitty: add option `theme`

### PR DESCRIPTION
**Pull request by NelsonJeppesen**
### Description
Add a new option to use the recently added `kitty-themes` nix package to set the theme. Use the friendly name of the theme seen in `kitty +kitten themes` 

https://github.com/kovidgoyal/kitty-themes

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).
The current test for kitty is a simple match against a example file. Because this change adds a path to the nix store, that will no longer work because the path can change. I'm not sure if it's worth re-writing all the kitty tests just to cover themes

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [ ] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: modules/programs/kitty.nix [hunk 1]
```diff
@@ -72,6 +72,18 @@ in {
...
     };
 
+    theme = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = ''
+        Apply a Kitty color theme. This option takes the friendly name of
+        any theme given by the command <literal>kitty +kitten themes</literal>.
+        See <link xlink:href="https://github.com/kovidgoyal/kitty-themes"/>
+        for more details
```
Comment (sumnerevans) [inline modules/programs/kitty.nix:82]:
```suggestion
        for more details.
```

File: modules/programs/kitty.nix [hunk 2]
```diff
@@ -72,6 +72,18 @@ in {
       '';
     };
 
+    theme = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = ''
+        Apply a Kitty color theme. This option takes the friendly name of
+        any theme given by the command <literal>kitty +kitten themes</literal>.
```
Comment (sumnerevans) [inline modules/programs/kitty.nix:80]:
```suggestion
        any theme given by the command <command>kitty +kitten themes</command>.
```

**Review by sumnerevans — CHANGES_REQUESTED** [review]

**Review by sumnerevans — APPROVED** [review]

**Comment by kryachkov** [comment]
Fails with
```
error: attribute 'kitty-themes' missing, at /home/andfalk/.nix-defexpr/channels/home-manager/modules/programs/kitty.nix:146:21
```

when I add `theme = "Solarized Light";` to my `home.nix`. Any ideas how to fix?

**Comment by NelsonJeppesen** [comment]
> Fails with
> ```
> error: attribute 'kitty-themes' missing, at /home/andfalk/.nix-defexpr/channels/home-manager/modules/programs/kitty.nix:146:21
> ```
> 
> when I add `theme = "Solarized Light";` to my `home.nix`. Any ideas how to fix?
> 

I believe kitty themes is only on NixOS unstable but will be available when 22.05 comes out. Are you running unstable or 21.11?

**Comment by kryachkov** [comment]
Yaha, correct, I'm on 21.11. Will wait for 22.05 release then. Thanks

**Pull request merged**